### PR TITLE
connected Categories to Products

### PR DESCRIPTION
**Pull request by AdamVojtus**
in Products added new parameter Category
In webAPI added as well

### CODE REVIEW

File: Eshop.Domain/Product.cs [hunk 1]
```diff
@@ -2,36 +2,40 @@
...
+        public void Update(string title, string description, decimal price, Category category)
         {
-            ValidateParameters(title, description, price);
+            ValidateParameters(title, description, price, category);
 
             Title = title;
             Description = description;
             Price = price;
+            Category = category;
+
```
Comment (akonaweb) [inline Eshop.Domain/Product.cs:35]: remove vertical space

File: Eshop.Domain/Product.cs [hunk 2]
```diff
@@ -2,36 +2,40 @@
...
+            ValidateParameters(title, description, price, category);
 
             Id = id;
             Title = title;
             Description = description;
             Price = price;
+            Category = category;
         }
 
         public int Id { get; private set; }
```
Comment (akonaweb) [inline Eshop.Domain/Product.cs:21]: remove from Id private set; - check video to understand why

File: Eshop.WebApi/Controllers/ProductController.cs [hunk 3]
```diff
@@ -7,10 +7,16 @@ namespace Eshop.WebApi.Controllers
...
+        {
+            new Category(1, "Computers"),
+            new Category(2, "Mouses")
+        };
+
         private static List<Product> Products = new List<Product>
         {
-            new Product(1, "Notebook Acer 16", "Best notebook out there", 399.99m),
-            new Product(2, "Mouse Razor 123", "Best Mouse", 14.50m),
+            new Product(1, "Notebook Acer 16", "Best notebook out there", 399.99m, Categories[1]),
```
Comment (akonaweb) [inline Eshop.WebApi/Controllers/ProductController.cs:18]: Category[0], you see we have no tests and you did not test it properly so it is a bug.

File: Eshop.Domain/Product.cs (hunk 1 above)
Comment (AdamVojtus) [reply Eshop.Domain/Product.cs:35]: fixed

File: Eshop.WebApi/Controllers/ProductController.cs (hunk 3 above)
Comment (akonaweb) [reply Eshop.WebApi/Controllers/ProductController.cs:18]: You have not commited it. It is probably only on your local. You can see it also in commit - you can not win this.
Check it properly.

File: Eshop.Domain/Product.cs (hunk 2 above)
Comment (AdamVojtus) [reply Eshop.Domain/Product.cs]: fixed properly

**Review by akonaweb — APPROVED** [review]

**Pull request merged**